src/modules/lua/script_lua.c

```diff
@@ -189,451 +221,388 @@ void *luaGetFromRegistry(lua_State *lua, const char *name) {
-     * have a newline that needs to be trimmed. In any case the lua server error table shouldn't end with a newline. */
-    msg = sdstrim(msg, "\r\n");
-    sds final_msg = sdscatfmt(error_code, " %s", msg);
-
-    lua_newtable(lua);
-    lua_pushstring(lua, "err");
-    lua_pushstring(lua, final_msg);
-    lua_settable(lua, -3);
-
-    sdsfree(msg);
-    sdsfree(final_msg);
-}
-
-void luaPushError(lua_State *lua, const char *error) {
-    luaPushErrorBuff(lua, sdsnew(error));
+    return s;
 }
 
-/* In case the error set into the Lua stack by luaPushError() was generated
- * by the non-error-trapping version of server.pcall(), which is server.call(),
- * this function will raise the Lua error so that the execution of the
- * script will be halted. */
-int luaError(lua_State *lua) {
-    return lua_error(lua);
+char *copy_string_from_lua_stack(lua_State *lua) {
+    const char *str = lua_tostring(lua, -1);
+    size_t len = lua_strlen(lua, -1);
+    char *res = ValkeyModule_Alloc(len + 1);
+    strncpy(res, str, len);
+    res[len] = 0;
+    return res;
 }
 
-
-/* ---------------------------------------------------------------------------
- * Lua reply to server reply conversion functions.
- * ------------------------------------------------------------------------- */
-
 /* Reply to client 'c' converting the top element in the Lua stack to a
  * server reply. As a side effect the element is consumed from the stack.  */
-static void luaReplyToServerReply(client *c, client *script_client, lua_State *lua) {
+static void luaReplyToServerReply(ValkeyModuleCtx *ctx, int resp_version, lua_State *lua) {
     int t = lua_type(lua, -1);
 
     if (!lua_checkstack(lua, 4)) {
         /* Increase the Lua stack if needed to make sure there is enough room
          * to push 4 elements to the stack. On failure, return error.
          * Notice that we need, in the worst case, 4 elements because returning a map might
          * require push 4 elements to the Lua stack.*/
-        addReplyError(c, "reached lua stack limit");
+        ValkeyModule_ReplyWithError(ctx, "ERR reached lua stack limit");
         lua_pop(lua, 1); /* pop the element from the stack */
         return;
     }
 
     switch (t) {
-    case LUA_TSTRING: addReplyBulkCBuffer(c, (char *)lua_tostring(lua, -1), lua_strlen(lua, -1)); break;
+    case LUA_TSTRING:
+        ValkeyModule_ReplyWithStringBuffer(ctx, lua_tostring(lua, -1), lua_strlen(lua, -1));
+        break;
     case LUA_TBOOLEAN:
-        if (script_client->resp == 2)
-            addReply(c, lua_toboolean(lua, -1) ? shared.cone : shared.null[c->resp]);
-        else
-            addReplyBool(c, lua_toboolean(lua, -1));
+        if (resp_version == 2) {
+            int b = lua_toboolean(lua, -1);
+            if (b) {
+                ValkeyModule_ReplyWithLongLong(ctx, 1);
+            } else {
+                ValkeyModule_ReplyWithCString(ctx, NULL);
```

Comment:
This logic is so weird. It's preserving the logic though. It looks correct.

I believe this is equivalent though:

```suggestion
                ValkeyModule_ReplyWithNull(ctx);
```